discovery-provider/src/tasks/index_profile_challenge_backfill.py

```diff
@@ -0,0 +1,145 @@
+from src.models.social.save import Save
+from src.tasks.celery_app import celery
+from src.tasks.social_features import (
+    dispatch_challenge_follow,
+    dispatch_challenge_repost,
+)
+from src.tasks.user_library import dispatch_favorite
+from src.utils.config import shared_config
+from src.utils.prometheus_metric import save_duration_metric
+from src.utils.session_manager import SessionManager
+from src.utils.update_indexing_checkpoints import (
+    get_last_indexed_checkpoint,
+    save_indexed_checkpoint,
+)
+
+logger = logging.getLogger(__name__)
+
+index_profile_challenge_backfill_tablename = "index_profile_challenge_backfill"
+
+# Number of blocks to scan through at a time
+BLOCK_INTERVAL = 1000
+
+
+def enqueue_social_rewards_check(
+    db: SessionManager, challenge_bus: ChallengeEventBus, redis: Redis
+):
+    with db.scoped_session() as session:
+        block_backfill = get_latest_backfill(session)
+        if block_backfill is None:
+            logger.info("index_profile_challenge_backfill.py | No backfill block")
+            return
+        # Do it
+        max_blocknumber_seen = block_backfill
+        # reposts of tracks and playlists reposts
+        reposts: List[Repost] = (
+            session.query(Repost)
+            .filter(
+                Repost.blocknumber > block_backfill,
+                Repost.blocknumber <= block_backfill + BLOCK_INTERVAL,
+            )
+            .all()
+        )
+        for repost in reposts:
+            dispatch_challenge_repost(challenge_bus, repost, repost.blocknumber)
+            max_blocknumber_seen = max(repost.blocknumber, max_blocknumber_seen)
+
+        saves: List[Save] = (
+            session.query(Save)
+            .filter(
+                Save.blocknumber > block_backfill,
+                Save.blocknumber <= block_backfill + BLOCK_INTERVAL,
+            )
+            .all()
+        )
+        for save in saves:
+            dispatch_favorite(challenge_bus, save, save.blocknumber)
+            max_blocknumber_seen = max(save.blocknumber, max_blocknumber_seen)
+
+        follows: List[Follow] = (
+            session.query(Follow)
+            .filter(
+                Follow.blocknumber > block_backfill,
+                Follow.blocknumber <= block_backfill + BLOCK_INTERVAL,
+            )
+            .all()
+        )
+        for follow in follows:
+            dispatch_challenge_follow(challenge_bus, follow, follow.blocknumber)
+            max_blocknumber_seen = max(follow.blocknumber, max_blocknumber_seen)
+
+        save_indexed_checkpoint(
+            session, index_profile_challenge_backfill_tablename, max_blocknumber_seen
+        )
+
+
+def get_latest_backfill(session: Session) -> Optional[int]:
+    try:
+
+        checkpoint = get_last_indexed_checkpoint(
+            index_profile_challenge_backfill_tablename
+        )
+        BACKFILL_SOCIAL_REWARDS_BLOCKNUMBER = (
+            shared_config["discprov"]["backfill_social_rewards_blocknumber"]
```

Comment:
will we have to coordinate the docker compose change before telling SPs to upgrade?

Reply:
we should make the change in the same PR as we update the tag and should be good

Reply:
we should be able to auto update this env